simpleviewinc/keg-core
# Use style callback example

**Ticket**: [ZEN-351](https://jira.simpleviewtools.com/browse/ZEN-351)

**Notes**
* This branch depends on [this PR](https://github.com/simpleviewinc/re-theme/pull/26)
* Either update the package.json to pull from that branch
* Or checkout that branch for your local re-theme, and use the keg-cli to run a docker container and sync the repo
  * Example =>
    * `keg core`
    * `keg pr 63`
    * `cd ~/keg/re-theme`
    * `keg pr 26`
    * `keg core start --sync retheme`
    * `keg core att`
    * `cd $DOC_RETHEME_PATH`
    * `yarn install; yarn build`

## Context
* Events force is defining their styles in CSS
* Their clients can define custom CSS as well
* Currently, the tap, using re-theme, is only compatible with JSON style objects
* We need to come up with a solution to interfacing with CSS
* It also needs to be able to update styles if the css updates

## Goal
* To fix issues with import re-theme and keg-compoents
* To show a demo of the re-theme hook `useStylesCallback`

## Updates
* Added `.dockerignore`
* `core/base/components/kegComponents.js`
  * Fixed import
* `core/base/containers/homeContainer.js`
  * Added `useStylesCallback` example - **The Important Part** 
* `core/configs/webpack.config.js`
  * Added direct path resolution to a number of repos
  * This allows us to install `node_modules` for sub packages like `re-theme`
    * Normally you would run into an issue with 2 different react versions
    * By doing this, when running in the `keg-core`, all imports of react, will use the `keg-core` version
  * This has no effect on then running a sub-module like `re-theme`
* `package.json`
  * Added nodemon to `web` script, and added another script `tap:start`, which runs in a docker container
* `App.js`
  * Fixed imports of Dimensions, and Platform
    * Not really sure how it was working before

## Testing

**Test 1**
* **Note** If you already did the tests of this [re-theme PR](https://github.com/simpleviewinc/re-theme/pull/26), then you already tested running taps
* Run the docker package to test the tap =>
  * `keg dpg run --package docker.pkg.github.com/simpleviewinc/keg-packages/tap:zen-290-use-expo-vector-icons --command tap:start`
  * Ensure everything works as expected

**Test 2**
* **Note** If you already did the tests of this [re-theme PR](https://github.com/simpleviewinc/re-theme/pull/26), then you already tested running keg-core
* Run the docker package to test keg-core =>
  * `keg dpg run --package docker.pkg.github.com/simpleviewinc/keg-packages/keg-core:usestylecallback-example`
  * Ensure everything works as expected

**Test 3**
* Connect to the running `keg-core` container
* Run `keg dpg run --package docker.pkg.github.com/simpleviewinc/keg-packages/keg-core:usestylecallback-example --cmd test`
  * Ensure all tests pass
